### PR TITLE
Add error check to detect_clearsky

### LINKED ISSUE #2005: More descriptive errors for `detect_clearsky`
**Is your feature request related to a problem? Please describe.**
Using a window length that is too short relative to data period in `detect_clearsky` produces cryptic errors

**Describe the solution you'd like**
Raise a ValueError that directly explains the problem

Example:
```
import pvlib
import pandas as pd
import numpy as np
start = '2012-01-01'
end = '2015-01-01'
freq = '60T'
times = pd.date_range(start=start, end=end, freq=freq)
x1 = pd.Series(np.random.rand(len(times)), index=times)
x2 = pd.Series(np.random.rand(len(times)), index=times)

pvlib.clearsky.detect_clearsky(x1, x2,
                          window_length=90, mean_diff=75, max_diff=75,
                          lower_line_length=-45, upper_line_length=80,
                          var_diff=0.032, slope_dev=75)
```
Results in the following:
```
ValueError                                Traceback (most recent call last)
Cell In[28], line 11
      8 x1 = pd.Series(np.random.rand(len(times)), index=times)
      9 x2 = pd.Series(np.random.rand(len(times)), index=times)
---> 11 pvlib.clearsky.detect_clearsky(x1, x2,
     12                           window_length=90, mean_diff=75, max_diff=75,
     13                           lower_line_length=-45, upper_line_length=80,
     14                           var_diff=0.032, slope_dev=75)

File ~/opt/anaconda3/envs/rdtools3_testing/lib/python3.10/site-packages/pvlib/clearsky.py:854, in detect_clearsky(measured, clearsky, times, infer_limits, window_length, mean_diff, max_diff, lower_line_length, upper_line_length, var_diff, slope_dev, max_iterations, return_components)
    850 clear_line_length = _line_length_windowed(
    851     scaled_clear, H, samples_per_window, sample_interval)
    853 line_diff = meas_line_length - clear_line_length
--> 854 slope_max_diff = _max_diff_windowed(
    855     meas - scaled_clear, H, samples_per_window)
    856 # evaluate comparison criteria
    857 c1 = np.abs(meas_mean - alpha*clear_mean) < mean_diff

File ~/opt/anaconda3/envs/rdtools3_testing/lib/python3.10/site-packages/pvlib/clearsky.py:602, in _max_diff_windowed(data, H, samples_per_window)
    600 def _max_diff_windowed(data, H, samples_per_window):
    601     raw = np.diff(data)
--> 602     raw = np.abs(raw[H[:-1, ]]).max(axis=0)
    603     return _to_centered_series(raw, data.index, samples_per_window)

File ~/opt/anaconda3/envs/rdtools3_testing/lib/python3.10/site-packages/numpy/core/_methods.py:40, in _amax(a, axis, out, keepdims, initial, where)
     38 def _amax(a, axis=None, out=None, keepdims=False,
     39           initial=_NoValue, where=True):
---> 40     return umr_maximum(a, axis, None, out, keepdims, initial, where)

ValueError: zero-size array to reduction operation maximum which has no identity
```

And the following produces a different error, even though the `window_length` > period as required by the docstring (Does the window length need to be a multiple of the period?)
```
import pvlib
import pandas as pd
import numpy as np
start = '2012-01-01'
end = '2015-01-01'
freq = '60T'
times = pd.date_range(start=start, end=end, freq=freq)
x1 = pd.Series(np.random.rand(len(times)), index=times)
x2 = pd.Series(np.random.rand(len(times)), index=times)

pvlib.clearsky.detect_clearsky(x1, x2,
                          window_length=175, mean_diff=75, max_diff=75,
                          lower_line_length=-45, upper_line_length=80,
                          var_diff=0.032, slope_dev=75)
```
result:
```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Cell In[32], line 11
      8 x1 = pd.Series(np.random.rand(len(times)), index=times)
      9 x2 = pd.Series(np.random.rand(len(times)), index=times)
---> 11 pvlib.clearsky.detect_clearsky(x1, x2,
     12                           window_length=175, mean_diff=75, max_diff=75,
     13                           lower_line_length=-45, upper_line_length=80,
     14                           var_diff=0.032, slope_dev=75)

File ~/opt/anaconda3/envs/rdtools3_testing/lib/python3.10/site-packages/pvlib/clearsky.py:891, in detect_clearsky(measured, clearsky, times, infer_limits, window_length, mean_diff, max_diff, lower_line_length, upper_line_length, var_diff, slope_dev, max_iterations, return_components)
    885     except AttributeError:
    886         message = "Optimizer exited unsuccessfully: \
    887                    No message explaining the failure was returned. \
    888                    If you would like to see this message, please \
    889                    update your scipy version (try version 1.8.0 \
    890                    or beyond)."
--> 891     raise RuntimeError(message)
    893 else:
    894     alpha = optimize_result.x

RuntimeError: Optimizer exited unsuccessfully: NaN result encountered.
```

### PR DESCRIPTION
**Pull request by cwhanse**
- [x] Closes #2005
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Tests added
 - [x] Updates entries in [`docs/sphinx/source/reference`](https://github.com/pvlib/pvlib-python/blob/main/docs/sphinx/source/reference) for API changes.
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/main/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels (including `remote-data`) and Milestone are assigned to the Pull Request and linked Issue.

Adds a check on window_length in `clearsky.detect_clearsky` to ensure window_length contains at least 3 data points.

### CODE REVIEW

File: pvlib/clearsky.py [hunk 1]
```diff
@@ -812,6 +816,12 @@ def detect_clearsky(measured, clearsky, times=None, infer_limits=False,
     sample_interval, samples_per_window = \
         tools._get_sample_intervals(times, window_length)
 
+    if samples_per_window < 3:
+        raise ValueError(f"Window of length {window_length} found"
+                         " Each window must contain at least"
+                         " three data points; {samples_per_window} results"
+                         " from time interval of {sample_interval}")
```
[the remark this answers is not in the document]
Comment (kandersolar) [reply pvlib/clearsky.py:823]: The portion of the message after the semicolon could probably be clearer.  It wasn't immediately obvious to me how to interpret `2 results from time interval of 1.0`.

File: pvlib/clearsky.py [hunk 2]
```diff
@@ -812,6 +816,15 @@ def detect_clearsky(measured, clearsky, times=None, infer_limits=False,
...
         tools._get_sample_intervals(times, window_length)
 
+    if samples_per_window < 3:
+        raise ValueError(f"Samples per window of {samples_per_window}"
+                         f" found. Each window must contain at least 3 data"
+                         f" points."
+                         " Samples per window is calculated from window_length"
+                         " ({window_length} found) and the data interval"
+                         f" ({sample_interval} found)."
+                         " Increase window_length.")
```
Comment (kandersolar) [inline pvlib/clearsky.py:826]:
```suggestion
        raise ValueError(f"Samples per window of {samples_per_window}"
                         " found. Each window must contain at least 3 data"
                         " points."
                         " Samples per window is calculated from window_length"
                         f" ({window_length} found) and the data interval"
                         f" ({sample_interval} found)."
                         " Increase window_length.")
```

**Review by kandersolar — APPROVED** [review]

File: pvlib/clearsky.py [hunk 3]
```diff
@@ -812,6 +816,15 @@ def detect_clearsky(measured, clearsky, times=None, infer_limits=False,
...
         tools._get_sample_intervals(times, window_length)
 
+    if samples_per_window < 3:
+        raise ValueError(f"Samples per window of {samples_per_window}"
+                         " found. Each window must contain at least 3 data"
+                         " points."
+                         " Samples per window is calculated from window_length"
+                         f" ({window_length} found) and the data interval"
+                         f" ({sample_interval} found)."
+                         " Increase window_length.")
```
Comment (adriesse) [inline pvlib/clearsky.py:826]: Could you provide the minimum window length?

**Comment by cwhanse** [comment]
Adding a bit of scope here. #1784 added a function to lookup thresholds from Jordan and Hansen (2023). That paper extended the detect_clearsky algorithm to plane-of-array (POA) data. #1784 didn't update the docstring to describe the new capability.

**Comment by kandersolar** [comment]
Thanks @cwhanse!

**Pull request merged**